### PR TITLE
.obj .mtl export

### PR DESCRIPTION
**Pull request by wlwlw**

### CODE REVIEW

File: WallsAndHoles/simpletexturedobject.cpp [hunk 1]
```diff
@@ -157,3 +157,31 @@ const QImage &SimpleTexturedObject::getImage() const
...
+
+SharedImageAndSource SimpleTexturedObject::getImageAndSource() const
+{
+    return mImage;
+}
+
+QString SimpleTexturedObject::getMaterialName() const
+{
+    QString feature = QString::number(getAmbient()+getDiffuse()+getSpecular()+getShininess())+getImageAndSource()->source();
+    return QString(QCryptographicHash::hash(feature.toUtf8(),QCryptographicHash::Md5).toHex());
```
Comment (timoffex) [inline WallsAndHoles/simpletexturedobject.cpp:186]: Interesting solution!

File: WallsAndHoles/objtools.h [hunk 2]
```diff
@@ -241,5 +246,115 @@ inline bool saveOBJSimpleTextured(QString path, const SimpleTexturedObject &obje
...
+        // Vertex List
+        QVector<Vertex> vertices;
+        // Index List
+        QVector<unsigned int> indices;
+
+        // Material
+        QString materialName;
+    };
+}
+*/
```
Comment (timoffex) [inline WallsAndHoles/objtools.h]: Is this necessary to keep, or did you mean to remove it?

**Review by timoffex — COMMENTED** [review]
First pass -- everything looks good and compiles! I can't figure out how to get Preview (on Mac) to actually show the textures when I export though.

File: WallsAndHoles/objtools.h [hunk 3]
```diff
@@ -241,5 +246,115 @@ inline bool saveOBJSimpleTextured(QString path, const SimpleTexturedObject &obje
...
+typedef QSharedPointer<Material> SharedMaterial;
+
+class OBJModel
+{
+public:
+    OBJModel(QString _name="map"){
+        name = _name;
+        mSaveDirectory = '.';
+    }
+    ~OBJModel(){
```
Comment (Bdtrotte) [inline WallsAndHoles/objtools.h]: Not required

File: WallsAndHoles/teapot.obj [hunk 4]
```diff
@@ -1,3 +1,5 @@
+mtllib spider.mtl
```
Comment (Bdtrotte) [inline WallsAndHoles/teapot.obj]: Lets remove

File: WallsAndHoles/simpletexturedobject.cpp (hunk 1 above)
Comment (Bdtrotte) [inline WallsAndHoles/simpletexturedobject.cpp:186]: This needs to be the name the user sets... If the fact that the user could have different materials with the same name is a problem, then handle that, and add "_id" to the end of the name. Currently, when importing to something like unity, you end up with a bunch of materials whos name makes no sense...

**Review by Bdtrotte — CHANGES_REQUESTED** [review]

**Pull request merged**